src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -147,8 +147,10 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
             // This is e.g. an "extern" method in C# without a DllImport or InternalCall.
             if (methodIL == null)
             {
-                ThrowHelper.ThrowInvalidProgramException(ExceptionStringID.InvalidProgramSpecific, MethodBeingCompiled);
+                methodNoIL = true;
```

Comment:
I would really like `CompileMethodInternal` to stay true to the "compile method or throw" contract. Bool flags make it hard to reason about what's going on.

I would:
* Delete the `_isFallbackBodyCompilation` bool flag from here. There's one more reference to the flag in `initClass` - put that in the `#else` branch of `#if READYTORUN`. We can put the definition of `_isFallbackBodyCompilation` on the AOT side. R2R doesn't use it.
* Change signature of `CompileMethodInternal` to only take a single `MethodIL` parameter (`methodCodeNeedingCode` is redundant with `_methodCodeNode`). The `MethodIL` parameter will become mandatory. Keep the throw for null MethodIL in place to keep the contract (CompileMethodInternal either compiles or throws). If you don't want the throw to trigger, just check for it in `CompileMethod`. `CompileMethod` will now have to provide the IL.
* I would also update `Get_CORINFO_METHOD_INFO` to return a bool instead of MethodIL. We no longer need the `MethodIL` after that and returning `MethodIL` is an odd API.